base/stream.jl

```diff
@@ -876,6 +894,14 @@ function stop_reading(stream::LibuvStream)
     end
 end
 
+function readbytes!(s::LibuvStream, b::AbstractArray{UInt8}, nb=length(b))
+    wait_readnb(s, nb)
+    nr = nb_available(s)
+    resize!(b, nr) # shrink to just contain input data if was resized
+    read!(s.buffer, b)
```

Comment:
I'm not sure why something should be an `IO` type if it doesn't have `read` and `write`.